### PR TITLE
[keymap] superfell's custom keymap for the 0xc pad

### PR DESCRIPTION
**Pull request by superfell**
Adding my custom keymap for the s_ol/0xc_pad 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/s_ol/0xc_pad/keymaps/superfell/config.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
```
Comment (waffle87) [inline keyboards/s_ol/0xc_pad/keymaps/superfell/config.h:16]: This file should be removed.

File: keyboards/s_ol/0xc_pad/keymaps/superfell/keymap.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                TO(_NUMPAD), QK_BOOT,
+               XXXXXXX, XXXXXXX, XXXXXXX,
+           RGB_MOD, XXXXXXX, XXXXXXX, XXXXXXX,
+              RGB_TOG, RGB_VAD, RGB_VAI
+    ),
+};
+// clang-format on
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (biton32(state)) {
```
Comment (waffle87) [inline keyboards/s_ol/0xc_pad/keymaps/superfell/keymap.c:46]:
```suggestion
    switch (get_highest_layer(state)) {
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by superfell** [comment]
Thanks, I've made the changes you noted.

**Review by waffle87 — APPROVED** [review]

File: keyboards/s_ol/0xc_pad/keymaps/superfell/keymap.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define _NUMPAD 0
+#define _CONTROL 1
+#define _ADJUST 2
```
Comment (drashna) [inline keyboards/s_ol/0xc_pad/keymaps/superfell/keymap.c:20]:
```suggestion
enum custom_layers {
    _NUMPAD,
    _CONTROL,
    _ADJUST,
};
```

File: keyboards/s_ol/0xc_pad/keymaps/superfell/readme.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+![0xC.pad Layout Image](https://i.imgur.com/pk4G33B.png)
```
Comment (drashna) [inline keyboards/s_ol/0xc_pad/keymaps/superfell/readme.md:1]:
```suggestion
![0xC.pad Layout Image](https://i.imgur.com/pk4G33Bh.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**